src/main/java/de/jtheb/fh/lagerverwaltung/entities/Compartment.java

```diff
@@ -42,19 +42,26 @@ public int getItemsVolume() {
         return volume;
     }
 
-    public boolean itemFits(Item item) {
-        this.getArticleNr();
+    public boolean itemFitsWithArticleNr(Item item) {
+        return (this.articleNrFits(item) && itemFitsVolume(item));
+    }
+
+    public boolean itemFitsVolume(Item item) {
         return (this.getVolume() - this.getItemsVolume() - item.getVolume() >= 0);
     }
 
     public boolean isFull() {
         return (this.getVolume() == this.getItemsVolume());
     }
 
-    public String getArticleNr() {
+    public String getArticleNr() { //TODO Check in correct return null
         for (Item item : items) {
             return item.getArticleNr();
         }
         return null;
```

Comment:
add javadoc stating that null means no item was found. think about how you will handle non-assigned shelves.